pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/DescriptorLookupCache.java

```diff
@@ -112,11 +112,15 @@ public synchronized StepDescriptor lookupStepDescriptor(String n) {
         return describableMap.get(n);
     }
 
-    public synchronized Descriptor<? extends Describable> lookupStepOrFunction(String name) {
+    public synchronized Descriptor<? extends Describable> lookupStepFirstThenFunction(String name) {
```

Comment:
we should put `@Restricted(NoExternalUse.class)` on things so that I can feel more calm when you change method signatures like this. I believe you can put in on classes as well now.